### ISSUE TITLE
[mastodon/activitypub] Link "mastodon-style" direct messages to friendica messages

### ISSUE THREAD

**Issue by kr428** [friendica/friendica#12180]
Not sure whether these are the right terms, but with Mastodon and Pleroma users sending me "direct messages" via activitypub, I usually end up messing things up because I'm having a hard time finding these once I lost track of them somewhere. The native Friendica "message" feature seems to work in a different (and incompatible) way.
Would it be possible to handle activitypub direct messages just the way Friendica direct messages are handled...?

**Comment by annando** [comment]
In Mastodon there is no difference if a post is sent to a limited group of people or is meant as DM. In Pleroma there is one additional value - and we use that to display the message as DM. But as long as Mastodon doesn't do something like that we cannot distinct non public posts from DMs.

**Comment by kr428** [comment]
How does Mastodon make a difference here then? Is it just UI (and the fact that, by "accident", in this case there's only one recipient)? Wouldn't it be possible to replicate whatever Mastodon does here to make these "direct messages" appear in DMs? I mean,a fter all in Mastodon in some way I _do_ have this kind of distinction...?

**Comment by annando** [comment]
Mastodon doesn't make a difference. They simply display a post that mentions you in the mentions.

**Comment by kr428** [comment]
Maybe it would suffice to add this to "messages" or some dedicated view then? My biggest gripe with the current setup is that, once you have more notifications in your history, it's hard (not to say impossible) to filter out that particular messages from, say, public messages mentioning you. In Mastodon, for this purpose the "direct message" view usually works reasonably well. In Friendica, at the moment it's tough to impossible to track these down, in some cases the only way that helps is e-mail notifications. Or am I missing something obvious here?

**Comment by annando** [comment]
We have to completely rework our direct messages system, since not only the data model is really old but also it lacks several functionalities and has got many other flaws. I guess that there is no quick fix for this issue.

**Comment by kr428** [comment]
Ok, I see. Would it be / is it possible to search for such messages via user interface in some way? Like "find messages mentioning me and marked as private"?

**Comment by loma-one** [comment]
@kr428 I don't know exactly what your goal is. However, I personally would prefer DMs to continue to be displayed as DMs in a fixed location rather than embedding them in a general stream of messages in the personal timeline.
The messages (DM) are my fixed anchor where I would expect to see all private messages, no matter what network they come from.

**Comment by kr428** [comment]
@loma-one I personally do see two goals here: For long-term expectation, I'm completely with you, I'd expect to see all DMs I received or sent in the Friendica "messages" view. But I have no idea whether / how this is even possible. Short-term goal, thus, would be two-sided:

1. Have an obvious way to send a private "direct message" to a user on some Fediverse instance no matter which platform is used on the receiving end.
2. Have an obvious way of quickly finding "direct messages" I received from a Mastodon/Firefish/ActivityPub user in Friendica.

(1), as far as I tested yesterday (using Mentions and visibility "only for <user>", seems to somehow work but I'm unsure if there are any caveats as the messages sent out this way display on Mastodon as "follower-only", not "private"). For (2), I only have workarounds (such as keeping mail notifications for Mastodon-style "private mention DMs" in an IMAP folder).

**Comment by xundeenergie** [comment]
I tried the new Flohmarkt https://codeberg.org/grindhold/flohmarkt and tried to send a comment to a posting and got this as answer

![1132222459654ef3e6d3da1886300377-1](https://github.com/friendica/friendica/assets/5894829/027f456e-ebe3-4482-8a25-ce14aeb42901)

So, to use flohmarkt from friendica, i must send a comment with permission "for mentioned only", which is not possible now in friendica.

**Comment by annando** [comment]
@xundeenergie You can do so via the API.

**Comment by kr428** [comment]
> @xundeenergie You can do so via the API.

Yes, that generally works but also is a bit itchy at times, seems Mastodon API in Friendica also has a bunch of rough edges. Right now, then again, there are just few features remaining that require a dedicated mobile app for using Friendica. And maybe getting the PWA there seems a better approach?

**Comment by xundeenergie** [comment]
> @xundeenergie You can do so via the API.

How can i do that?

**Comment by kr428** [comment]
> > @xundeenergie You can do so via the API.
> 
> How can i do that?

Use an app like tusky, fedilab or elk.zone. 🙂

**Comment by xundeenergie** [comment]
> > > @xundeenergie You can do so via the API.
> > 
> > 
> > How can i do that?
> 
> Use an app like tusky, fedilab or elk.zone. 🙂

hmmm... i only use webapp...

**Comment by kr428** [comment]
> 
> hmmm... i only use webapp...

Webapp apparently can't handle this as of now. 😔 That's one reason why this issue exists...

**Comment by kr428** [comment]
Closing this off. Seems to have improved a bit in recent versions or just work better and guess that's fine for now.
